### PR TITLE
add dbscan algorithm for clustering.

### LINKED ISSUE #330: add new algorithm for jubaclustering
Jubaclustering has two clustering algorithm, K-means and GMM.
Both algorithm needs the number of clusters before deviding into cluster.

I suggest to add new algorithm which does not need the number of clusters for Jubaclustering.

### PR DESCRIPTION
**Pull request by yukimori**
fix #330 

add dbscan algorithm for jubaclustering.

dbscan algorithm has two parameter eps and min_core_point.
It does not need k parameter, because it does not calculate center point of the cluster.

This algorithm does not support (throw unsupported_method exception)
- get_k_center
- get_nearest_center
- get_nearest_members_light

### CODE REVIEW

File: jubatus/core/clustering/dbscan.cpp [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+std::vector<wplist> dbscan::get_clusters() const {
+  return clusters;
+}
+
+void dbscan::set_eps(double eps) {
+  eps_ = eps;
+}
+
+// return cluster as wplist
+wplist dbscan::expand_cluster(size_t idx, wplist& points) {
```
Comment (TkrUdagawa) [inline jubatus/core/clustering/dbscan.cpp]: ``idx`` and ``points`` are const.

File: jubatus/core/clustering/dbscan.cpp [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+namespace jubatus {
+namespace core {
+namespace clustering {
+
+dbscan::dbscan(double eps, size_t min_core_point)
+    : eps_(eps),
+      min_core_point_(min_core_point) {
+}
+
+void dbscan::batch(wplist& points) {
```
Comment (TkrUdagawa) [inline jubatus/core/clustering/dbscan.cpp]: ``points`` is const.

File: jubatus/core/clustering/dbscan.cpp [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        }
+      }
+      core = added_core;
+    } while(!core.empty());
+  }
+  return cluster;
+}
+
+
+std::vector<size_t> dbscan::region_query(size_t idx, wplist& points) {
```
Comment (TkrUdagawa) [inline jubatus/core/clustering/dbscan.cpp]: ``idx`` and ``points`` are const

File: jubatus/core/clustering/dbscan_clustering_method.cpp [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  throw JUBATUS_EXCEPTION(core::common::unsupported_method("get_nearest_center_index"));
+}
+
+common::sfv_t dbscan_clustering_method::get_nearest_center(
+    const common::sfv_t& point) const {
+  throw JUBATUS_EXCEPTION(core::common::unsupported_method("get_nearest_center"));
+}
+
+wplist dbscan_clustering_method::get_cluster(
+    size_t cluster_id,
```
Comment (TkrUdagawa) [inline jubatus/core/clustering/dbscan_clustering_method.cpp]: ``cluster_id`` is also const.

**Comment by TkrUdagawa** [comment]
In addition, please fix cpplint errors.

**Comment by TkrUdagawa** [comment]
Great Work!
Please fix the documents for dbscan.

File: jubatus/core/clustering/clustering_config.hpp [hunk 5]
```diff
@@ -72,7 +78,9 @@ struct clustering_config {
         & JUBA_MEMBER(compressed_bucket_size)
         & JUBA_MEMBER(forgetting_factor)
         & JUBA_MEMBER(forgetting_threshold)
-        & JUBA_MEMBER(seed);
+        & JUBA_MEMBER(eps)
+        & JUBA_MEMBER(seed)
+        & JUBA_MEMBER(min_core_point);
```
Comment (kazuki) [inline jubatus/core/clustering/clustering_config.hpp]: @yukimori @TkrUdagawa 
This comment is important for usability.
Do you plan to fix this point for 1.0 release?

Comment (TkrUdagawa) [inline jubatus/core/clustering/clustering_config.hpp]: Thanks for the comment.
We are fixing the problem in #334, #335.

Comment (kazuki) [inline jubatus/core/clustering/clustering_config.hpp]: Thanks!

**Pull request merged**